test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartDirectMethodReport.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator.Reports.EdgeHubRestartTest
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    /// <summary>
+    /// This is a counting report to show test result counts, e.g. expect and match counts; and contains a list of unmatched test results.
+    /// </summary>
+    class EdgeHubRestartDirectMethodReport : TestResultReportBase
+    {
+        public EdgeHubRestartDirectMethodReport(
+            string trackingId,
+            string resultType,
+            bool isPassing,
+            long passedMessageCount,
+            Dictionary<string, ulong> messageCount,
+            Dictionary<HttpStatusCode, List<TimeSpan>> completedStatusHistogram,
+            TimeSpan minPeriod,
+            TimeSpan maxPeriod,
+            TimeSpan medianPeriod,
+            TimeSpan meanPeriod,
+            TimeSpan variancePeriod)
+            : base(trackingId, resultType)
+        {
+            this.isPassing = isPassing;
+            this.PassedMessageCount = passedMessageCount;
```

Comment:
why only have passedMessageCount? but no failed message count?  Please define TotalCallsInvoked and TotalCallsReceived; and remove isPassing; which isPassed should be determined by TotalCallsInvoked and TotalCallsReceived.

Reply:
The `TotalCallsInvoked` and `TotocallsReceived` are tracked by the dictionary `messageCount`. The passedMessageCount is another metric that counts how many of the messages are "good" independent of the previous messages.